Sources/Core/JavaModelRenderer.swift

```diff
@@ -16,54 +16,170 @@ public struct JavaModelRenderer: JavaFileRenderer {
+            self.className + " that = (" + self.className + ") o;",
+            "return " + bodyEquals + ";"
+            ]
+        }
     }
 
-    func renderBuilderProperties(modifiers: JavaModifier = [.public, .abstract]) -> [JavaIR.Method] {
+    func renderModelProperties(modifiers: JavaModifier = [.private]) -> [JavaIR.Property] {
         let props = self.transitiveProperties.map { param, schemaObj in
-            JavaIR.method(modifiers, "Builder set\(param.snakeCaseToCamelCase())(\(self.typeFromSchema(param, schemaObj)) value)") {[]}
+            JavaIR.Property(annotations: ["SerializedName(\"\(param)\")"], modifiers: [.private], type: self.typeFromSchema(param, schemaObj), name: param.snakeCaseToPropertyName(), initialValue: "")
+        }
+
+        let bits = JavaIR.Property(annotations: [], modifiers: [.private], type: "int", name: "_bits", initialValue: "0")
+
+        var bitmasks: [JavaIR.Property] = []
+        var index = 0
+        self.transitiveProperties.forEach { param, _ in
+            bitmasks.append(JavaIR.Property(annotations: [], modifiers: [.private, .static, .final], type: "int", name: param.uppercased() + "_SET", initialValue: "1 << " + String(index)))
+            index += 1
         }
-        return props
+
+        return props + bitmasks + [bits]
+    }
+
+    func renderModelGetters(modifiers: JavaModifier = [.public]) -> [JavaIR.Method] {
+        let getters = self.transitiveProperties.map { param, schemaObj in
+            JavaIR.method(modifiers, self.typeFromSchema(param, schemaObj) + " get" + param.snakeCaseToCapitalizedPropertyName() + "()") {[
+                "return this." + param.snakeCaseToPropertyName() + ";"
+                ]}
+        }
+        return getters
     }
 
-    func renderModelProperties(modifiers: JavaModifier = [.public, .abstract]) -> [JavaIR.Method] {
-        return self.transitiveProperties.map { param, schemaObj in
-            JavaIR.method(modifiers, "@SerializedName(\"\(param)\") \(self.typeFromSchema(param, schemaObj)) \(param.snakeCaseToPropertyName())()") {[]}
+    func renderModelIsSetCheckers(modifiers: JavaModifier = [.public]) -> [JavaIR.Method] {
+        let getters = self.transitiveProperties.map { param, _ in
+            JavaIR.method(modifiers, "boolean get" + param.snakeCaseToCapitalizedPropertyName() + "IsSet()") {[
+                "return (this._bits & " + param.uppercased() + "_SET) == " + param.uppercased() + "_SET;"
+            ]}
         }
+        return getters
     }
 
-    func renderTypeClassAdapter() -> JavaIR.Method {
-        return JavaIR.method([.public, .static], "TypeAdapter<\(className)> jsonAdapter(Gson gson)") {[
-            "return new AutoValue_\(className).GsonTypeAdapter(gson);"
+    func renderModelMergeWith() -> JavaIR.Method {
+        return JavaIR.method([.public], self.className + " mergeWith(" + self.className + " model)") {[
+            self.className + ".Builder builder = this.toBuilder();",
```

Comment:
I can switch the whole thing over if you feel strongly, but I really found it hard to read/code particularly because of the need to include a "\". I spoke to some in-house iOS folks who seemed to feel that concatenation through + was fine.

Reply:
I think as long as we're not executing functions inside `\()`, for simple strings this is a bit easier to read but that's just my opinion. I'd prefer that they were using string interpolation for simple strings